libpod/stats.go

```diff
@@ -56,7 +56,11 @@ func (c *Container) GetContainerStats(previousStats *define.ContainerStats) (*de
 
 	previousCPU := previousStats.CPUNano
 	now := uint64(time.Now().UnixNano())
+	stats.Duration = cgroupStats.CPU.Usage.Total
+	stats.UpTime = time.Duration(stats.Duration)
 	stats.CPU = calculateCPUPercent(cgroupStats, previousCPU, now, previousStats.SystemNano)
+	stats.AvgCPU = calculateAvgCPU(stats.CPU, previousStats.CPUTrack)
+	stats.CPUTrack = append(previousStats.CPUTrack, float64(cgroupStats.CPU.Usage.Total))
```

Comment:
any risk that this can grow indefinitely or use outdated info (or should the average computed using all the available data points)?

If we need all data points, could we just store the current average and number of data points to calculate the incremental average?

Reply:
Good idea, i will store number of datapoints and the current average I am pretty sure I can do `((oldAvg * numDataPoints) + newData) / (numDataPoints + 1)`